mlflow/pyfunc/scoring_server/client.py

```diff
@@ -0,0 +1,143 @@
+        if ping_status.status_code != 200:
+            raise Exception(f"ping failed (error code {ping_status.status_code})")
+
+    def wait_server_ready(self, timeout=30):
+        begin_time = time.time()
+        while True:
+            time.sleep(0.3)
+            try:
+                self.ping()
+                return
+            except Exception:
+                pass
+            if time.time() - begin_time > timeout:
+                break
+        raise RuntimeError('Wait scoring server ready timeout.')
+
+    def get_module_version(self, module_name):
+        """
+        Get module version on the scoring server worker python environment.
+        This method is for testing purpose.
+        """
+        status = requests.get(url=self.url_prefix + f"/version/{module_name}")
+        if status.status_code != 200:
+            raise Exception(f"get_module_version failed (error code {status.status_code})")
+        return status.text.strip()
+
+    def invoke(self, data, pandas_orient="records"):
+        import pandas as pd
+
+        content_type_list = []
+        if type(data) == pd.DataFrame:
+            content_type_list.append(scoring_server.CONTENT_TYPE_JSON)
+            if pandas_orient == "records":
+                content_type_list.append(scoring_server.CONTENT_TYPE_FORMAT_RECORDS_ORIENTED)
+            elif pandas_orient == "split":
+                content_type_list.append(scoring_server.CONTENT_TYPE_FORMAT_SPLIT_ORIENTED)
+            else:
+                raise Exception(
+                    "Unexpected pandas_orient for Pandas dataframe input %s" % pandas_orient
+                )
+        else:
+            raise RuntimeError("Unsupported data type.")
+
+        post_data = json.dumps(scoring_server._get_jsonable_obj(data, pandas_orient=pandas_orient))
+
+        response = requests.post(
+            url=self.url_prefix + "/invocations",
+            data=post_data,
+            headers={"Content-Type": "; ".join(content_type_list)},
+        )
+
+        if response.status_code != 200:
+            raise Exception(
+                f"Invocation failed (error code {response.status_code}, response: {response.text})"
+            )
+
+        return scoring_server.load_predictions_from_json_str(response.text)
+
+
+def prepare_env(local_model_path, stdout=sys.stdout, stderr=sys.stderr):
+    cmd = [
+        "mlflow", "models", "prepare-env", "-m", local_model_path,
+        "--install-mlflow",
+    ]
+    return subprocess.run(
+        cmd,
+        stdout=stdout,
+        stderr=stderr,
+        universal_newlines=True,
+        check=True
+    )
+
+
+def start_server(
+        server_port, local_model_path,
+        host='127.0.0.1', num_workers=1,
+        no_conda=False,
+        env=None, stdout=sys.stdout, stderr=sys.stderr,
+):
+    cmd = [
+        "mlflow", "models", "serve", "-m", local_model_path,
+        "-h", host,
+        "-p", str(server_port), "-w", str(num_workers), "--install-mlflow",
```

Comment:
Can we remove `--install-mlflow` or not?

Reply:
No. We need install the current PR version mlflow in the created environment. Because I add some code in the scoring server module.

The conda yaml includes a mlflow dependency which results in installing the latest released version mlflow.

Reply:
> We need install the current PR version mlflow in the created environment.

But users don't need to, right?

Reply:
Oh yes user don't need to. let me update code